src/main/java/com/shoppinglist/shoppinglist/dao/IShoppingListDAO.java

```diff
@@ -6,7 +6,7 @@
 import java.util.NoSuchElementException;
 
 public interface IShoppingListDAO {
-    ListItem save(ListItem item);
+    ListItem saveItem(ListItem item);

```

Comment:
I changed save to saveItem to make it more descriptive and clearly express its purpose. In order to avoid conflicts with other saving functions that may be developed in the future
https://www.tutorialspoint.com/java/java_documentation.htm
https://rhamedy.medium.com/a-short-summary-of-java-coding-best-practices-31283d0167d3
https://www.baeldung.com/java-clean-code